### PR TITLE
Bump to 0.7.0

### PR DESCRIPTION
**Pull request by jsha**
And bump dependencies from alphas to released versions, fix examples to accommodate breaking changes.

### CODE REVIEW

**Comment by ctz** [comment]
Note we should avoid "=" version specs. Those are only useful for pre-releases, and are actively bad for public releases.

**Comment by cpu** [comment]
> cpu force-pushed the release-0.7.0 branch from 8277c2d to 0f55cc4 

* Added the pki-types update to fix the build breakage.
* Updated the examples to accommodate rustls 0.22 API changes to fix the tests.
* Removed the `=` version specs for the updated deps that are no longer pinned to alphas.

**Comment by cpu** [comment]
Hm I think we should also probably separate out the crate version update into its own commit.

**Comment by cpu** [comment]
> separate out the crate version update into its own commit. 

Done.

File: Cargo.toml [hunk 1]
```diff
@@ -11,16 +11,16 @@ repository = "https://github.com/rustls/rustls-native-certs"
 categories = ["network-programming", "cryptography"]
 
 [dependencies]
-rustls-pemfile = "=2.0.0-alpha.2"
-pki-types = { package = "rustls-pki-types", version = "0.2.2" }
+rustls-pemfile = "2.0.0"
```
Comment (djc) [inline Cargo.toml:14]: I think we usually prefer to admit trailing zero components.

File: src/lib.rs [hunk 2]
```diff
@@ -73,12 +73,18 @@ fn load_certs_from_env() -> Option<Result<Vec<CertificateDer<'static>>, Error>>
...
                 format!("could not load PEM file {path:?}: {err}"),
             )
-        })
+        })?;
+
+    let certs: Vec<CertificateDer<'static>> = certs
+        .into_iter()
+        .map(|c| c.into_owned())
+        .collect();
+    Ok(certs)
```
Comment (djc) [inline src/lib.rs:89]: All this reduces to

```rust
    rustls_pemfile::certs(&mut f)
        .map(|result| match result {
            Ok(der) => Ok(der),
            Err(err) => Err(Error::new(
                ErrorKind::InvalidData,
                format!("could not load PEM file {path:?}: {err}"),
            )),
        })
        .collect()
```

File: Cargo.toml (hunk 1 above)
Comment (cpu) [reply Cargo.toml]: I removed the existing instances in an initial commit, and then added updates that adhere to this style afterwards.

File: src/lib.rs (hunk 2 above)
Comment (cpu) [reply src/lib.rs]: Nice :-) Tidied up with your suggestion.

**Comment by djc** [comment]
Hmm, should we avoid the `collect()` in `load_pem_certs()` and yield an iterator like we do in rustls-pemfile?

**Comment by jsha** [comment]
> 
> Hmm, should we avoid the collect() in load_pem_certs() and yield an iterator like we do in rustls-pemfile?

Tempting. It would be nice to have symmetry between the two packages. Just now I tried going down this road a little ways:

```rust
fn load_pem_certs(
    path: &Path,
) -> impl Iterator<Item = Result<CertificateDer<'static>, Error>> + '_ {
    let mut f = BufReader::new(File::open(path)?);
    rustls_pemfile::certs(&mut f).map(|result| match result {
        Ok(der) => Ok(der),
        Err(err) => Err(Error::new(
            ErrorKind::InvalidData,
            format!("could not load PEM file {path:?}: {err}"),
        )),
    })
}
```

That first `?` is a problem because this function doesn't return `Result`, it returns `impl Iterator<Item = Result<_>>`. The natural thing to do from an implementation perspective is to return `Result<impl Iterator<Item = Result<CertificateDer<'static>, Error>>, Error>`. But from an API design perspective that's bad; we'd prefer to reduce the depth at which we nest results and the places at which we can return error.

I tried turning that first potential error into an `iter::once` containing an error, but then we get conflicting iterator types: `Once` vs `Map`. We could fix that by boxing up the return value, but I don't like that either.

My slight inclination is to leave the return type for `load_pem_certs()` as a `Vec`. There's some missed optimization - someone could map the result with [`webpki::anchor_from_trusted_cert`](https://docs.rs/rustls-webpki/latest/webpki/fn.anchor_from_trusted_cert.html) and thus quickly deallocating each Vec rather than storing it. But I think that doesn't really amount to much.

Thanks to all for the fixups! This looks good to me.

**Comment by djc** [comment]
Agreed.

**Pull request merged**